sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +599,190 @@ def test_param_sampler():
+                                           cv=n_folds, iid=True,
+                                           param_distributions=params)
+    random_search_iid.fit(X, y)
+
+    param_keys_gs = ('param_C', 'param_degree', 'param_gamma', 'param_kernel')
+    param_keys_rs = ('param_C', 'param_gamma')
+    score_keys = ('test_accuracy_mean',
+                  'test_accuracy_rank',
+                  'test_split_0_accuracy',
+                  'test_split_1_accuracy',
+                  'test_split_2_accuracy',
+                  'test_accuracy_std')
+
+    all_searches = (grid_search, grid_search_iid,
+                    random_search, random_search_iid)
+    all_param_keys = (param_keys_gs,) * 2 + (param_keys_rs,) * 2
+    all_n_candidates = (n_grid_points,) * 2 + (n_search_iter,) * 2
+    is_iid = [False, True] * 2
+
+    # Common tests
+    for search, param_keys, n_cand, iid in zip(all_searches, all_param_keys,
+                                               all_n_candidates, is_iid):
+        results = search.results_
+
+        # Test array type and dtype
+        assert_true(all(isinstance(results[param], np.ma.MaskedArray)
+                        for param in param_keys))
+        assert_true(all(results[key].dtype is np.dtype('O')
+                        for key in param_keys))
+        assert_false(any(isinstance(results[key], np.ma.MaskedArray)
+                         for key in score_keys))
+        assert_true(all(results[key].dtype is np.dtype('float64')
+                        for key in score_keys if not key.endswith('rank')))
+
+        # Test it contains all the required results
+        assert_array_equal(sorted(six.iterkeys(results)),
+                           sorted(param_keys + score_keys))
+        assert_true(all(results[key].shape[0] == n_cand
+                        for key in param_keys + score_keys))
+
+        # SVC is a classifier, StratifiedKFold is used
+        cv = StratifiedKFold(n_folds)
+        res_scores = np.vstack([results["test_split_%d_accuracy" % i]
+                                for i in range(n_folds)]).T
+        res_means = results["test_accuracy_mean"]
+        res_stds = results["test_accuracy_std"]
+        res_ranks = results["test_accuracy_rank"]
+        test_sample_counts = ([len(test) for _, test in cv.split(X, y)]
+                              if iid else None)
+        assert_equal(iid, search.iid)
+        assert_equal(iid, (test_sample_counts is not None))
+        means = np.average(res_scores, axis=1, weights=test_sample_counts)
+        stds = np.sqrt(np.average(
+            (res_scores - means[:, np.newaxis]) ** 2,
+            axis=1, weights=test_sample_counts))
+        assert_array_equal(res_means, means)
+        assert_array_equal(res_stds, stds)
+
+        ranks = rankdata(means, method='max')
+        # Invert the ranks. NOTE that method='max' becomes 'min' when inverted.
+        ranks = (n_cand + 1) - ranks
+        best_index = np.flatnonzero(ranks == 1)[0]
```

Comment:
`np.argmax / argmin` should have the same behaviour, if you want alternate methods of testing